lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts

```diff
@@ -412,14 +433,190 @@ describe('ContentMetadataComponent', () => {
 
             fixture.detectChanges();
             tick(100);
-            clickOnSave();
+            clickOnGeneralInfoSave();
 
             tick(100);
             expect(component.node).toEqual(expectedNode);
             expect(updateService.updateNodeAspect).toHaveBeenCalledWith(expectedNode);
         }));
     });
 
+    describe('toggleEdit', () => {
+        let mockEvent: MouseEvent;
+        let showErrorSpy: jasmine.Spy;
+        const mockGroup: CardViewGroup = {
+            editable: false,
+            expanded: false,
+            title: '',
+            properties: []
+        };
+        beforeEach(() => {
+            mockEvent = new MouseEvent('click');
+            component.editableGroup = mockGroup;
+            showErrorSpy = spyOn(notificationService, 'showError').and.stub();
+        });
+
+        it('should toggle General Info editing mode', () => {
+            component.editable = false;
+            component.toggleEdit(mockEvent, mockGroup, ButtonType.GeneralInfo);
+            expect(component.editableTags).toBe(false);
+            expect(component.editableCategories).toBe(false);
+            expect(component.editableGroup.editable).toBe(false);
+        });
+
+        it('should toggle Tags editing mode', () => {
+            component.editableTags = false;
+            component.toggleEdit(mockEvent, mockGroup, ButtonType.Tags);
+            expect(component.isTagPanelVisible).toBe(component.editableTags);
+            expect(component.tagNameControlVisible).toBe(true);
+            expect(component.editableCategories).toBe(false);
+            expect(component.editableGroup.editable).toBe(false);
+        });
+
+        it('should toggle Categories editing mode', () => {
+            component.editableCategories = false;
+            component.toggleEdit(mockEvent, mockGroup, ButtonType.Categories);
+            expect(component.isCategoriesPanelVisible).toBe(component.editableCategories);
+            expect(component.categoryControlVisible).toBe(true);
+            expect(component.editableTags).toBe(false);
+            expect(component.editableGroup.editable).toBe(false);
+        });
+
+        it('should toggle Group editing mode', () => {
+            component.toggleEdit(mockEvent, mockGroup, ButtonType.Group);
+            expect(component.editable).toBe(false);
+            expect(component.editableGroup).toBe(mockGroup.editable ? mockGroup : null);
+            if (mockGroup.editable) {
```

Comment:
Why is this `if` condition here? You should now what is expected and expect those condition to appear

Reply:
Updated the test case.